zerver/views/realm.py

```diff
@@ -262,6 +265,20 @@ def update_realm(
             )
             data["notifications_stream_id"] = notifications_stream_id
 
+    if reportmsg_stream_id is not None:
+        if realm.reportmsg_stream is None or (realm.reportmsg_stream.id != reportmsg_stream_id):
+            new_reportmsg_stream = None
+            if reportmsg_stream_id >= 0:
+                try:
+                    new_reportmsg_stream = get_stream_by_id_in_realm(reportmsg_stream_id, realm)
```

Comment:
Why is this not `access_stream_by_id`? For security reasons, we always want to use `access_` functions to access data, but I can imagine a product reason we might want to allow specifying a private stream the administrator can't access... still, if that's the intent, we can pass `allow_realm_admin=True`.

Reply:
DONE